src/wasm/wasm-type.cpp

```diff
@@ -227,8 +265,46 @@ template<typename Self> struct TypeGraphWalkerBase {
   // This base walker does not know when to stop scanning, so at least one of
   // these needs to be overridden with a method that calls the base scanning
   // method only if some end condition isn't met.
-  void scanType(Type* type);
-  void scanHeapType(HeapType* ht);
+  void scanType(Type* type) {
+    if (type->isBasic() || type->isRef()) {
+      return;
+    }
+    if (type->isTuple()) {
+      auto& types = const_cast<Tuple&>(type->getTuple());
+      for (auto it = types.rbegin(); it != types.rend(); ++it) {
+        taskList.push_back(Task::scan(&*it));
+      }
+      return;
```

Comment:
What is this line achieving? The function ends right after it.

Reply:
Nothing, this can be cleaned up. This was the result of mindlessly turning the switch with `break`s into `if`s.

Reply:
Ok, let me read it again then. I was just wondering if I wasn't looking at this code right at all.

Reply:
It turns out there's a lot about the type graph walker that can be simplified. I think it had to support much more complicated usage in the past. Will push an update with these cleanups and more.